### ISSUE TITLE
irc/discord/slack/telegram

### ISSUE THREAD

**Issue by MadcapJake** [howl-editor/howl#210]
Is there a chatroom where howl users/devs congregate? If not, perhaps there should be one (any preference for a service to use?)

**Comment by kirbyfan64** [comment]
Well, we just set up a [subreddit](https://www.reddit.com/r/howleditor/), though I don't quite think that's what you meant...

What about Gitter?

**Comment by MadcapJake** [comment]
Gitter is fine too!

**Comment by abaez** [comment]
I think either gitter or telegram should be the choices. Leaning more so to telegram. Mostly due to it being _slightly_ more transparent and open than gitter.

**Comment by abaez** [comment]
Also, I found this little tool for telegram recently and think it works pretty well: [integram][1].

[1]: https://integram.org/

**Comment by shalabhc** [comment]
Telegram didn't let me create an account because I use a Google Voice phone number. Is there any way around that?

**Comment by MadcapJake** [comment]
Seems that Telegram doesn't support VoIP numbers at all (or they did at one point but don't anymore).

**Comment by abaez** [comment]
On Sat, Jun 4, 2016, at 11:32 PM, Shalabh Chaturvedi wrote:
> Telegram didn't let me create an account because I use a Google Voice
> phone number. Is there any way around that?
 
I don't think they do. I'm no telegram expert though. Only offering an
alternative to the options of an IM, for the editor's group at large.
 
 Still, I think it is worth considering. Just looking briefly at the
 api, we could probably get say a bundle to interop, with the cli
 version of telegram, to have some weird integration with howl.

**Comment by Phrohdoh** [comment]
IRC is as open as these tools get.

**Comment by shalabhc** [comment]
Agreed - I like IRC too. It's definitely easy to get started and has no prerequisites like sending your phone number to a third party ;-)

Also, I already have a registered nick on freenode.

**Comment by abaez** [comment]
Personally speaking, only brought up telegram to get the discussion flowing on the subject. Having both IM and forum is an excellent way to preserving an active community for any tool. I think the votes on @Phrohdoh comment speak for themselves. IRC is the way to go. It has worked this long for a reason.  :)

**Comment by kirbyfan64** [comment]
FWIW IRC doesn't maintain any kind of log of messages by default, which kind of sucks when someone had some sort of "really important thing" and you want to see it.

**Comment by shalabhc** [comment]
A logbot could solve the logging issue. But I haven't looked into how much work it is to run and maintain one , and if there are any logbot services out there.

**Comment by abaez** [comment]
Here's a small list of what so far has become a desire:

``` yaml
keywords syntax:
  accesible = can newcomers invite themselves?
  mod       = moderate or modify the chat
  open      = is the client open source
  log       = can search the logs of conversation
  pin       = can pin a topic or post
  mobile    = can it be used on mobile
  
Enum types for answer:
  yes
  no
  _         = do not know

============================================================
client    accesible     mod     open    log     pin   mobile
------------------------------------------------------------
irc       yes           yes     yes     yes(4)  yes   yes(1)
telegram  yes           yes     yes(5)  no(2)   yes   yes
slack     no            yes     no      yes(3)  yes   yes
discord   yes           no      no      no(2)   no    yes
gitter    yes           yes     no      yes     _     yes    




1: works on mobile, but battery life is severely limited.
2: unless you define scrolling up in the conversation as logging.
3: logs, but only the first 10k comments. Rest is paywall.
4: logs, but you need to use an outside source.
5: only the api for extensibility with _promise_ of open source later on...
```

**Comment by kirbyfan64** [comment]
Gitter also has the advantage of GitHub integration, so you just sign in with your GitHub account (no need for another account) and can do stuff like reference issues. In addition, you can set it that new builds on Travis also get sent to Gitter.

**Comment by abaez** [comment]
If we going with integration, slack can do all of that (excluding new account) to the umpteenth level. Still good to know. :+1:

**Comment by nilnor** [comment]
All right, so we have some different alternatives proposed. In order to move forward I'd like to narrow these down to two: IRC or Gitter. 

- IRC does not need much of an introduction - it's been around for a long time, it's open, and lot's of people have experience with it. Drawbacks are no automatic archiving, no images or fancy stuff, and decidedly sub-par experience on mobile. 

- Gitter, features support for images, automatic archives, code snippets, Git integration, etc., but is a closed web thing.

(The above are just some points, see @abaez matrix for more complete evaluation points).

Let's choose one of the two, so if you have any preference leave a comment saying so and we'll settle on the winner (for now at least).

**Comment by abaez** [comment]
Put me for IRC. If the time is taken to add things to it, we could probably get all the bells and whistles that exist on gitter. Heck, just look at [ipfs] and their irc integration to [matrix].

[ipfs]: https://github.com/ipfs/ipfs
[matrix]: https://matrix.org/docs/projects/try-matrix-now.html

**Comment by kirbyfan64** [comment]
My vote goes to Gitter...if it weren't already obvious. :)

Also, remember that some IRC clients are closed-source, so "open-ness" isn't really that much of a deciding factor.

**Comment by MadcapJake** [comment]
For those looking for an excellent IRC experience on mobile check out [IRCCloud](https://www.irccloud.com) (Although this is a closed-source client, it's really quite nice).

My vote is for IRC over Gitter mostly because the Gitter linux client runs very poorly on my machine. Also though some IRC clients are closed-source, it's important to remember that the entire Gitter ecosystem is closed.

**Comment by shalabhc** [comment]
Heh, check out https://irc.gitter.im/

While I like IRC and am completely new to Gitter, I'm leaning towards Gitter after playing with it a bit. The reason is the *automatic* and painless integration with github as well as logging and search. Of course we could set up all of that with IRC, but do we want to support all of that?  Not to mention we also get markdown mode in messages and the ability to have a nice web link to the searchable chatroom from the website.

The only downside is the messages are stored in Gitter and not exportable. However with IRC, the messages aren't stored at all. If we can set up an IRC logger, we could perhaps set up the same logger with the IRC bridge linked above.

**Comment by shalabhc** [comment]
One more advantage I see with Gitter is seamless switching between devices. I currently switch between 3-4 devices and with IRC I lose the context, have to go to another site to look at the logs to catch up, and also have to kick off my nick or use different nicks.
